tools/image-tool/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    env,
+    path::{Path, PathBuf},
+    rc::Rc,
+};
+
+use async_trait::async_trait;
+use crane::CraneCLI;
+use docker::DockerCLI;
+use snafu::{ensure, ResultExt};
+use tokio::process::Command;
+use which::which;
+
+mod crane;
+mod docker;
+
+#[async_trait]
+pub trait ImageToolImpl {
+    /// Get the path of the cli tool to call
+    fn path(&self) -> PathBuf;
+
+    /// Pull an image archive to disk
+    async fn pull_archive(&self, file: &Path, uri: &str, cache: &Path) -> Result<()>;
+    /// Fetch the manifest
+    async fn get_manifest(&self, uri: &str) -> Result<Vec<u8>>;
+    /// Fetch the image config
+    async fn get_config(&self, uri: &str) -> Result<Vec<u8>>;
+
+    async fn output(&self, args: &[&str], error_msg: String) -> Result<Vec<u8>> {
+        let output = Command::new(&self.path())
+            .args(args)
+            .output()
+            .await
+            .context(error::CommandFailedSnafu { message: error_msg })?;
+        ensure!(
+            output.status.success(),
+            error::OperationFailedSnafu {
+                message: String::from_utf8_lossy(&output.stderr)
+            }
+        );
+        Ok(output.stdout)
+    }
+
+    async fn spawn(&self, args: &[&str], error_msg: String) -> Result<()> {
+        let status = Command::new(&self.path())
+            .args(args)
+            .spawn()
+            .context(error::CommandFailedSnafu {
+                message: error_msg.clone(),
+            })?
+            .wait()
+            .await
+            .context(error::CommandFailedSnafu {
+                message: error_msg.clone(),
+            })?;
+        ensure!(
+            status.success(),
+            error::OperationFailedSnafu {
+                message: error_msg.clone()
+            }
+        );
+        Ok(())
+    }
+}
+
+pub type ImageTool = Rc<dyn ImageToolImpl>;
+
+/// Auto-select the container tool to use by environment variable
+/// and-or auto detection
+pub fn image_tool() -> Result<ImageTool> {
+    if let Ok(name) = env::var("KIT_IMAGE_TOOL") {
+        return match name.as_str() {
+            "docker" => Ok(Rc::new(DockerCLI {
+                path: which("docker").context(error::NotFoundSnafu { name: "docker" })?,
+            })),
+            "crane" => Ok(Rc::new(CraneCLI {
```

Comment:
We should allow the other tools from the `crane` family:
```suggestion
            "crane" | "krane" | "gcrane" => Ok(Rc::new(CraneCLI {
```

Reply:
DO they all have the same cli command set?

Reply:
> Do they all have the same cli command set?

I believe that `crane` and `krane` do, and `gcrane` has additional commands on top of the shared set.